src/main/java/com/conveyal/analysis/GenerateRegionalAnalysisResults.java

```diff
@@ -0,0 +1,64 @@
+package com.conveyal.analysis;
+
+import com.conveyal.analysis.controllers.RegionalAnalysisController;
+import com.conveyal.analysis.models.OpportunityDataset;
+import com.conveyal.analysis.models.RegionalAnalysis;
+import com.conveyal.analysis.persistence.Persistence;
+import com.conveyal.file.FileStorage;
+import com.conveyal.file.FileStorageFormat;
+import com.conveyal.r5.analyst.progress.ProgressListener;
+import com.conveyal.r5.analyst.progress.TaskAction;
+import com.mongodb.DBObject;
+import com.mongodb.QueryBuilder;
+import org.mongojack.DBCursor;
+
+import java.util.Objects;
+
+public class GenerateRegionalAnalysisResults implements TaskAction {
+    private final FileStorage fileStorage;
+    public GenerateRegionalAnalysisResults (FileStorage fileStorage) {
+        this.fileStorage = fileStorage;
+    }
+
+    @Override
+    public void action(ProgressListener progressListener) throws Exception {
+        DBObject query = QueryBuilder.start().or(
+                QueryBuilder.start("travleTimePercentiles").is(null).get(),
+                QueryBuilder.start("cutoffsMinutes").is(null).get(),
+                QueryBuilder.start("destinationPointSetIds").is(null).get()
+        ).get();
+        try (DBCursor<RegionalAnalysis> cursor = Persistence.regionalAnalyses.find(query)) {
+            while (cursor.hasNext()) {
+                RegionalAnalysis regionalAnalysis = cursor.next();
+                int[] percentiles = Objects.requireNonNullElseGet(regionalAnalysis.travelTimePercentiles, () -> new int[]{regionalAnalysis.travelTimePercentile});
+                int[] cutoffs = Objects.requireNonNullElseGet(regionalAnalysis.cutoffsMinutes, () -> new int[]{regionalAnalysis.cutoffMinutes});
+                String[] destinationPointSetIds = Objects.requireNonNullElseGet(regionalAnalysis.destinationPointSetIds, () -> new String[]{regionalAnalysis.grid});
+
+                // Save as modern types
+                regionalAnalysis.cutoffsMinutes = cutoffs;
+                regionalAnalysis.travelTimePercentiles = percentiles;
+                regionalAnalysis.destinationPointSetIds = destinationPointSetIds;
+                Persistence.regionalAnalyses.put(regionalAnalysis);
```

Comment:
#956 mentions a MongoDB migration, and it looks like this PR is intended to perform that migration (via Java code) while generating the missing single-cutoff results files. If so we might want to mention that in #956 for future reference.

Reply:
Modified the description of #956 to reference this PR and recommend that this one should be merged and run first.